src/rp2_common/pico_unique_id/unique_id.c

```diff
@@ -20,8 +20,16 @@ static void __attribute__((constructor)) _retrieve_unique_id_on_boot(void) {
         // debug, so just produce something well-defined and obviously wrong.
         for (int i = 0; i < PICO_UNIQUE_BOARD_ID_SIZE_BYTES; i++)
             retrieved_id.id[i] = 0xee;
-    #else
+    #elif (PICO_UNIQUE_BOARD_ID_SIZE_BYTES == FLASH_UNIQUE_ID_SIZE_BYTES)
         flash_get_unique_id(retrieved_id.id);
+    #elif (PICO_UNIQUE_BOARD_ID_SIZE_BYTES < FLASH_UNIQUE_ID_SIZE_BYTES)
+        // The flash id is >8 bytes (e.g. IS25LP016D) but we want to keep the
+        // pico unique board id as 8 bytes, just use the last 8 bytes which are likely to change
+        uint8_t flash_id[FLASH_UNIQUE_ID_SIZE_BYTES];
+        flash_get_unique_id(flash_id);
+        memcpy(retrieved_id.id, flash_id + FLASH_UNIQUE_ID_SIZE_BYTES - PICO_UNIQUE_BOARD_ID_SIZE_BYTES, PICO_UNIQUE_BOARD_ID_SIZE_BYTES);
+    #else
+        #error unique board id size is greater than flash unique id size
```

Comment:
Super-minor nitpick, but maybe the "id" in this line should be "ID" ? (to match with line 11)